MapboxMobileEvents/MMENSURLSessionWrapper.m

```diff
@@ -29,8 +27,10 @@ - (void)reconfigure:(MMEEventsConfiguration *)configuration {
 #pragma mark MMENSURLSessionWrapper
 
 - (void)processRequest:(NSURLRequest *)request completionHandler:(void (^)(NSData * _Nullable data, NSURLResponse * _Nullable response, NSError * _Nullable error))completionHandler {
+    NSURLSession *session = [NSURLSession sessionWithConfiguration:[NSURLSessionConfiguration defaultSessionConfiguration] delegate:nil delegateQueue:nil];
```

Comment:
Do you not need the delegate any more?

Reply:
I've tested this without the delegate and we're still getting our `URLSession:didReceiveChallenge:` delegate callback. Since the documentation states:

> Your URLSession object doesn’t need to have a delegate. If no delegate is assigned, a system-provided delegate is used, and you must provide a completion callback to obtain the data.

https://developer.apple.com/documentation/foundation/urlsessiondelegate

I didn't see a good reason to keep it around since this is the only delegate method we have at the moment.

Reply:
I'm curious why the `system-provided` delegate would continue to call through to our delegate method? How would it event know?

Reply:
You're right it doesn't make it to the delegate method in that case. I was testing on the wrong build... I'll add `self` back as the delegate. 🤦‍♂

Reply:
Watch for the retain cycles!

Reply:
@julianrex can we catch retain cycles with static analysis tools?